### PR TITLE
TEST-#7316: Run a subset of CI tests with python 3.10 and 3.11 on a scheduled basis.

### LINKED ISSUE #7316: Run a subset of CI tests for all Python versions that Modin has declared supported on a scheduled basis.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

Python 3.11 has been tested separately in https://github.com/modin-project/modin/pull/7317. It works.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7316 
- [x] tests ~added and~ passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by YarShev** [comment]
Can we now switch to testing with python 3.10?

**Comment by anmyachev** [comment]
> Can we now switch to testing with python 3.10?

I don't think so, we can easily start using python functionality that isn't in 3.9 and break the code.

**Comment by YarShev** [comment]
Maybe we should add weekly testing on python 3.10 to catch such breaking changes so as to be prepared for transition to python 3.10?

File: .github/actions/python-only/action.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ description: "Prepare the environment to run simple tasks"
 inputs:
   python-version:
     description: "Python version to install"
-    default: "3.9.x"
+    default: "3.9"
```
Comment (anmyachev) [inline .github/actions/python-only/action.yml:6]: This does the same thing - installs the latest version of python. Reworked according to the prevailing style.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ on:
       - setup.py
       - versioneer.py
   push:
+  schedule:
```
Comment (YarShev) [inline .github/workflows/ci.yml:20]: Should this run only on push to main?

**Comment by YarShev** [comment]
Why only a subset of CI tests? Which tests are missing?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (anmyachev) [reply .github/workflows/ci.yml:20]: I don't think so. The load on CI resources will be greater. If necessary, in the future we can simply increase the launch frequency (not once a week, but twice, for example).

**Comment by anmyachev** [comment]
> Why only a subset of CI tests? Which tests are missing?

For example, `test-fuzzydata`.

**Review by YarShev — APPROVED** [review]

**Pull request merged**